File: Demo Projects/Fragments_NEETRoots/app/src/main/java/com/example/fragments_neetroots/MainActivity.java
package com.example.fragments_neetroots;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

import com.example.fragments_neetroots.Fragments.FirstFragment;
import com.example.fragments_neetroots.Fragments.SecondFragment;

public class MainActivity extends AppCompatActivity {
        Button first,second;
        LinearLayout layout;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        first=findViewById(R.id.first);
        second=findViewById(R.id.second);
        layout=findViewById(R.id.linearLayout2);

        first.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FirstFragment firstFragment=new FirstFragment();
                FragmentTransaction fragmentTransaction=getSupportFragmentManager().beginTransaction();
                fragmentTransaction.replace(R.id.linearLayout2,firstFragment);
                fragmentTransaction.commit(); //to start the transaction
            }
        });

        second.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                SecondFragment Fragment=new SecondFragment();
                FragmentTransaction fragmentTransaction=getSupportFragmentManager().beginTransaction();
                fragmentTransaction.replace(R.id.linearLayout2,Fragment);
                fragmentTransaction.commit(); //to start the transaction
            }
        });
    }
}

File: Demo Projects/navigation_app/app/src/main/java/com/example/navigation_app/MainActivity.kt
package com.example.navigation_app

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.Fragment
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity() {
    lateinit var drawerLayout: DrawerLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        drawerLayout = findViewById(R.id.drawer_layout)
        val navView: NavigationView = findViewById(R.id.navView)

        val toggle = ActionBarDrawerToggle(this, drawerLayout, R.string.open, R.string.Close)
        drawerLayout.addDrawerListener(toggle)
        toggle.syncState()
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        navView.setNavigationItemSelectedListener { menuItem ->
            when (menuItem.itemId) {
                R.id.nav_home -> replaceFragment(HomeFragment())
                R.id.nav_message -> replaceFragment(MessagesFragment())
                R.id.nav_delete -> replaceFragment(DeleteFragment())
                R.id.nav_login -> replaceFragment(LoginFragment())
            }
            drawerLayout.closeDrawer(GravityCompat.START)
            true
        }

        // Set the default fragment to be displayed on app launch
//        if (savedInstanceState == null) {
//            replaceFragment(HomeFragment())
//            navView.setCheckedItem(R.id.nav_home) // Set the checked item in the navigation drawer
//        }
    }

    override fun onBackPressed() {
        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    private fun replaceFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction()
            .replace(R.id.content_frame, fragment)
            .commit()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

            if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
                drawerLayout.closeDrawer(GravityCompat.START)
            } else {
                drawerLayout.openDrawer(GravityCompat.START)
            }
            return true

        return super.onOptionsItemSelected(item)
    }
}
